src/main/java/org/myrobotlab/service/abstracts/AbstractPinEncoder.java

```diff
@@ -6,41 +6,41 @@
 import org.myrobotlab.service.interfaces.EncoderControl;
 import org.myrobotlab.service.interfaces.EncoderController;
 
-public class AbstractEncoder extends Service implements EncoderControl {
+public class AbstractPinEncoder extends Service implements EncoderControl {

```

Comment:
poor name for the class.  Amt and AS 203 encoders are not "pin" based.  They are typically I2C or SPI based.  That doesn't map to a concept of a Pin in reality.  more like a bus address instead, especially for the i2c variants.

Reply:
There was confusion I suspect between the interface and the classes - I think it was pretty incomplete.  It looks like I did the original check-in (dunno where from but i'm sure I copy/pasted or just guessed on the original class) - You in-turn did the first implementations of it.  Obviously, it needs refining.  Feel free to change the name back, and remove any pin references, I was only trying **not** to break your existing implementations.